### PR TITLE
Gravity generator Fix.

### LINKED ISSUE #5602: Power loss doesn't turn off grav gen
The gravity generator doesn't seem to be to turning off when power is shut down in the area. Fixing this would add a lot of roleplay and cool potential, as well as atmosphere and tactical power use.

### PR DESCRIPTION
**Pull request by Sindorman**
Fixes gravity generator and improves it slightly.

- Gravity generator now reacts to area power change. Hence when APC looses power generator starts to discharge and when APC gains generator charges back up. Fixes #5602

- Gravity generator no longer processes areas if there was no gravity change. Saving some CPU and iteration over station list. Also this stops from producing wrong messages of gravity gain/loss when none happened.

- Entering or loosing area without gravity now produces a notice message.

### CODE REVIEW

**Comment by Sindorman** [comment]
!review

File: html/changelogs/Sindorman-grav.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - bugfix: "Gravity generator now is affected by any power change in the area. Hence when APC looses power so does the gravity generator, and gains otherwise."
+  - tweak: "Gravity generator no longer process gravity change if no change happened. This will stop message appearing about gravity change when none happened."
+  - tweak: "Entering or loosing area without gravity now produces a notice message."
```
Comment (skull132) [inline html/changelogs/Sindorman-grav.yml]:
```suggestion
  - tweak: "Entering or exiting area without gravity now produces a notice message."
```
erm, how do you lose an area?

Comment (Sindorman) [reply html/changelogs/Sindorman-grav.yml]: Sorry I did not check. I mean Entering area without gravity or loosing gravity in the area

File: html/changelogs/Sindorman-grav.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - bugfix: "Gravity generator now is affected by any power change in the area. Hence when APC looses power so does the gravity generator, and gains otherwise."
+  - tweak: "Gravity generator no longer process gravity change if no change happened. This will stop message appearing about gravity change when none happened."
+  - tweak: "Entering the area without gravity or loosing gravity in the area now produces a notice message."
```
Comment (skull132) [inline html/changelogs/Sindorman-grav.yml]:
```suggestion
  - tweak: "Entering the area without gravity or losing gravity in the area now produces a notice message."
```

grammar >:L

**Pull request merged**